app/controllers/util/FormField.scala

```diff
@@ -164,7 +164,7 @@ object RadioButtonGroupField {
     def dependenciesError: Option[String] = radioButtonGroupField.dependencies
       .get(option)
       .flatMap(_.flatMap {
-        case textField: TextField         => TextField.validate(dependencies(textField.fieldId), textField)
+        case textField: TextField         => TextField.validate(dependencies(textField.fieldId), textField.copy(isRequired = true))
```

Comment:
Does this impact other text fields that are not required and left empty?

Reply:
I don't think so as it's only required for the radio button field. In this case, if the user selects "yes" then a dependent field is required. For the other TextField, we are directly calling the `TextField.validate` method.